src/Agent.Worker/TestResults/JunitResultReader.cs

```diff
@@ -0,0 +1,309 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using System.Xml;
+using Microsoft.VisualStudio.Services.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class JUnitResultReader : IResultReader
+    {
+        public JUnitResultReader(IExecutionContext executionContext)
+        {
+            _executionContext = executionContext;
```

Comment:
There isn't any logs, you will want some logs for debugging. :)
You will need HostContext to trace